### ISSUE TITLE
Formatting Bug With Custom Type When Multiple Values Are Being Formatted With Build Option 'USE_NATIVEFMT'

### ISSUE THREAD

**Issue by USAFrenzy** [USAFrenzy/Serenity_Logger#16]
This bug is specific to the Serenity build with ArgFormatter as its formatting backend.
- ```fmtlib``` and ```<format>``` backends are unaffected by this bug

ISSUE:
------------------------------------------------------------------------
- When formatting a custom type value, if you have multiple substitution brackets present using the custom type, it will format the first value encountered correctly, however, subsequent formatting ends up bypassing the custom parsing and errors on a custom specifier, such as '%' for example if '%' was being used as a flag encounter token, thinking it's parsing a native type instead of a custom type.
  - EXAMPLE: 
  ```cpp 
   formatter::format("{:%L} - {:%l}", someCustomVariable);
   ```
- This issue is most likely related to the padding issue when using the static formatter::format() & formatter::format_to() functions where if you're adding a padding format call alongside custom formatting calls, it doesn't format the padding but *WILL* format the custom types, assuming each custom type call is its own separate formatting call. If you format a custom type with multiple substitution brackets, the first bug is present on top of the padding bug.
  - EXAMPLE: 
  ```cpp
  formatter::format("{0:*^50}\n{1:%L} - {1:%N}", "Some Headline", someCustomVariable);
  ```
This bug is being tracked over in the ```ArgFormatter``` repo [here](https://github.com/USAFrenzy/ArgFormatter/issues/1) and [here](https://github.com/USAFrenzy/ArgFormatter/issues/2)

**Comment by USAFrenzy** [comment]
Fixed in [ArgFormatter v1.0.2](https://github.com/USAFrenzy/ArgFormatter/releases/tag/v1.0.2)

**Comment by USAFrenzy** [comment]
Should note that the main branch doesn't reflect this change, only the dev branch and latest release at this moment. The dev branch was using tag 1.0.2 which was being tested, removed the old tag and set tag 1.0.2 to be the hotfix release that ArgFormatter v1.0.2 now references - therefore the current Serenity release [v1.0.0.5-alpha](https://github.com/USAFrenzy/Serenity_Logger/releases/tag/v1.0.0.5-alpha) already reflects this update
